presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2019,15 +2024,43 @@ private PhysicalOperation createLookupJoin(JoinNode node,
                     spillEnabled && !buildOuter && partitionCount > 1,
                     singleStreamSpillerFactory);
 
-            context.addDriverFactory(
-                    buildContext.isInputDriver(),
-                    false,
-                    ImmutableList.<OperatorFactory>builder()
-                            .addAll(buildSource.getOperatorFactories())
-                            .add(hashBuilderOperatorFactory)
-                            .build(),
-                    buildContext.getDriverInstanceCount(),
-                    buildSource.getPipelineExecutionStrategy());
+            if (SystemSessionProperties.isEnableDynamicFiltering(context.getSession()) && node.getType() == INNER && node.getCriteria().size() > 0) {
+                List<Symbol> filterSymbols = node.getDynamicFilters().values().stream().collect(toImmutableList());
+                List<String> dynamicFilterSymbolNames = node.getDynamicFilters().keySet().stream().collect(toImmutableList());
+                List<Integer> filterChannels = ImmutableList.copyOf(getChannelsForSymbols(filterSymbols, buildSource.getLayout()));
+
+                OperatorFactory dynamicFilterSource = new DynamicFilterSourceOperator.DynamicFilterSourceOperatorFactory(
+                        buildContext.getNextOperatorId(),
+                        node.getId(),
+                        buildSource.getTypes(),
+                        filterChannels,
+                        dynamicFilterSymbolNames,
+                        dynamicFilterClientSupplier,
+                        node.getId().toString(),
+                        buildContext.getDriverInstanceCount().orElse(1));
+
+                context.addDriverFactory(
+                        buildContext.isInputDriver(),
+                        false,
+                        ImmutableList.<OperatorFactory>builder()
+                                .addAll(buildSource.getOperatorFactories())
+                                .add(dynamicFilterSource)
```

Comment:
nit: consider building this list above the `if`, to prevent duplicating the `context.addDriverFactory(...)` invocation, e.g.:
https://github.com/prestosql/presto/blob/972c66d15f385ee05bfae1b5d5f9900471cf4466/presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java#L2084-L2102

Reply:
Done